### PR TITLE
ci: Update module_name filter step

### PR DESCRIPTION
**Pull request by edmundmiller**
Created a small test script to prevent re-running into edge-cases and to quickly add new test cases. If we don't hate the idea I think we can wire it up to the javascript file to avoid any copy-paste errors.

### CODE REVIEW

File: .github/scripts/get_module_path.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+// https://github.com/nf-core/modules/pull/7075
+test_case_1 = [
+    "modules/nf-core/umicollapse/tests/main.nf.test",
+    "modules/nf-core/umicollapse/tests/main.nf.test.snap",
+    "modules/nf-core/umicollapse/tests/nextflow.config",
+    "modules/nf-core/umicollapse/tests/nextflow_PE.config",
+    "modules/nf-core/umicollapse/tests/nextflow_SE.config",
+    "modules/nf-core/umitools/dedup/tests/main.nf.test",
+    "modules/nf-core/umitools/dedup/tests/main.nf.test.snap",
```
Comment (mashehu) [inline .github/scripts/get_module_path.js:33]:
```suggestion
    "modules/nf-core/umitools/dedup/tests/main.nf.test.snap",
    "modules/nf-core/umitools/dedup/main.nf",
```

**Review by mashehu — DISMISSED** [review]

**Review by MatthiasZepper — DISMISSED** [review]
I have now only tested the script in isolation, but the test cases pass for me locally as well. Thanks for the solid work!

**Comment by MatthiasZepper** [comment]
Looking at the CI failure in this PR...can it be that it was only the missing bracket in the .map() function that caused all the issues?


```
      - name: Get module name
        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7
        id: module_names
        with:
          script: |
            return [...new Set(${{ steps.filter.outputs.modules_files }}
                .map(path => path
                .replace('tests/', '')
                .replace('modules/nf-core/', '')
                .split('/')
                .slice(0, 2)
                .filter(x => !x.startsWith('main.nf') && x !== 'tests' && x !== 'meta.yml' && x !== 'environment.yml')
                .join('/'))
              )
            ];
```

File: modules/nf-core/upd/tests/nextflow.config [hunk 2]
```diff
@@ -1,3 +1,3 @@
 process {
-    ext.args = { "--proband earlycasualcaiman --father hugelymodelbat --mother slowlycivilbuck --af-tag AF regions"  }
+    ext.args = { "--proband earlycasualcaiman --father hugelymodelbat --mother slowlycivilbuck --af-tag AF regions" }
```
Comment (MatthiasZepper) [inline modules/nf-core/upd/tests/nextflow.config:2]: Not sure how this ended up in the PR, but otherwise that looks good to me.

**Review by MatthiasZepper — APPROVED** [review]
I think, that will do. Nice work!

[Zizmor](https://woodruffw.github.io/zizmor) nags that _`steps.module_names.outputs.result` may expand into attacker-controllable code_, but I don't really see how that could be exploited, because the inputs must be a valid file path. And in the worst case, `nf-core modules lint` is just called with some invalid module.

**Comment by kenibrewer** [comment]
Just flagging that this PR is needed to allow #8003 to be merged.

**Comment by edmundmiller** [comment]
Closing — lint filter logic was rewritten on main with a simpler approach (filter by main.nf/snap suffix + regex replace). This 21-commit iteration is superseded.
